0000-settlement-engine/0000-settlement-engine.md

```diff
@@ -0,0 +1,391 @@
+- If the settlement engine receives an incoming settlement:
+  1. The settlement engine MUST acknowledge the settlement to the accounting system.
+  2. The accounting system MUST add the amount to the accounts receivable.
+
+### Bilateral communication
+
+Interledger connectors use a transport, such as HTTP or WebSockets, to send and receive ILP packets with peers.
+
+To send messages or requests to the peer's settlement engine instance, settlement engine implementations SHOULD proxy all requests through a service operated by the connector. The connector will forward messages to the peer's connector using the shared transport, which will then send them to its settlement engine for processing. The peer settlement engine's response to the request will then be sent back, across the two connectors, to the origin settlement engine.
+
+#### Usage with [ILP-over-HTTP](https://github.com/interledger/rfcs/blob/master/0035-ilp-over-http/0035-ilp-over-http.md)
+
+To send the message to the peer, the raw message should be encoded in the `data` field of an ILP packet and addressed to `peer.settle`. The connector that receives the packet should identify the peer and send it to the settlement engine instanced associated with that account.
+
+ILP packets intended for the peer to proxy to its settlement engine should be addressed to `peer.settle` and MUST NOT be forwarded to any subsequent connectors.
+
+Connectors MUST track which settlement engine a peer's account is associated with and ensure their messages are delivered to that particular settlement engine.
+
+### Motivation
+
+Settlement engines supercede the [Ledger Plugin Interface (LPIv2)](https://github.com/interledger/rfcs/blob/master/0024-ledger-plugin-interface-2/0024-ledger-plugin-interface-2.md), an earlier abstraction for settlement integrations with Interledger. The new model addresses these issues:
+
+1. Multi-account plugins required logic for handling ILP packets, increasing implementation complexity.
+2. Plugins bundled settlement and bilateral communication functionality, limiting composability with other transports.
+3. The JavaScript specific interface prevented connector implementations in other programming languages from using existing plugins.
+4. Plugins operated in the same process as the connector, which limited scaling the two systems independently.
+
+## Specification
+
+### Core Functionality
+
+All settlement engine implementations MUST functionally guarantee that **the sum of amounts a settlement engine instance is instructed to settle should eventually equal the sum of amounts the peer's settlement engine instance acknowledges receipt of.**
+
+Varying implementations, arrangements, and conditions (such as connectivity to the settlement system) may delay consistency of instructed and acknowledged settlements by an indeterminate amount of time.
+
+### Accounts and identifiers
+
+Each account MUST be identified by a unique, URL-safe string.
```

Comment:
We should define what is `URL-safe` like mentioning the [RFC](https://tools.ietf.org/html/rfc3986).

Reply:
Added here: https://github.com/interledger/rfcs/compare/5ea7eddaa084d7dfc1aa2af56340037771227491...579e27cf5209cf3cda4bd19f615f647e3fd48776#diff-990a915317f4e35578edfa70631bd07dR183